srcalc/src/main/java/gov/va/med/srcalc/service/DefaultCalculationService.java

```diff
@@ -97,16 +115,27 @@ public CalculationResult runCalculation(
     }
 
     @Override
+    @Transactional
     public VistaPatientDao.SaveNoteCode signRiskCalculation(
             CalculationResult result, String electronicSignature)
     {
+        // Note: we must save the note first because this is how we check the given
+        // signature code.
         final VistaPatientDao.SaveNoteCode returnCode = 
             fPatientDao.saveRiskCalculationNote(
                 result.getPatientDfn(), electronicSignature, result.buildNoteBody());
+
         if (returnCode == SaveNoteCode.SUCCESS)
         {
             final SignedResult signedResult = result.signed();
+            // Save to the DB first because the DB can rollback if VistA fails. (VistA
+            // can't rollback if DB fails.)
+            fResultsDao.persistSignedResult(signedResult);
```

Comment:
I don't really see why, the method contract says it throws a `DataAccessException` so calling code can do what it likes, including logging. (`DisplayResultsController` does in fact log it.)